### ISSUE TITLE
AlDente crashing on macOS 12.6

### ISSUE THREAD

**Issue by alessionossa** [AppHouseKitchen/AlDente-Charge-Limiter#1003]
**Describe the bug**
I get a crash after a couple of seconds I try to open AlDente on my MacBoon with macOS 12.6. It works fine on my other MacBook with macOS 13.4; however, I cannot update anymore the first one

**To Reproduce**
Open AlDente - it will crash.

**Expected behavior**
AlDente should not crash on launch 

**Screenshots**
/

**MacBook (please complete the following information):**
 - MacBook model: MacBook Pro (15-inch, 2016)
 - macOS version: 12.6.7 (21G651)
 - AlDente Pro
 - AlDente Version: 1.22 (40)
 - List of all enabled/disabled settings in AlDente:

**Additional context**
The beginning of the crash report is:
```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               AlDente [34777]
Path:                  /Applications/AlDente.app/Contents/MacOS/AlDente
Identifier:            com.apphousekitchen.aldente-pro
Version:               1.22 (40)
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2023-07-22 15:06:16.7192 +0200
OS Version:            macOS 12.6.7 (21G651)
Report Version:        12
Bridge OS Version:     3.0 (14Y910)
Anonymous UUID:        4920635C-70E1-F011-B2B0-63F8850CC8F6

Sleep/Wake UUID:       0E569DD9-AF8E-4F84-96C2-AC9260C84E23

Time Awake Since Boot: 250000 seconds
Time Since Wake:       563 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000002000000010
Exception Codes:       0x0000000000000001, 0x0000002000000010
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [34777]

VM Region Info: 0x2000000010 is not in any region.  Bytes after previous region: 132944801809  Bytes before following region: 105415677313008
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      mapped file                 10bc7d000-10bdf5000    [ 1504K] r--/rw- SM=COW  ...t_id=17846fe5
--->  GAP OF 0x5ffef420b000 BYTES
      MALLOC_NANO              600000000000-600008000000 [128.0M] rw-/rwx SM=PRV  

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libswiftCore.dylib            	    0x7ff8173410c9 swift::runtime::AccessSet::insert(swift::runtime::Access*, void*, void*, swift::ExclusivityFlags) + 73
1   libswiftCore.dylib            	    0x7ff817341322 swift_beginAccess + 66
2   SwiftUI                       	    0x7ff915c02c8d NSHostingView.checkForReentrantLayout() + 64
3   SwiftUI                       	    0x7ff915c09047 NSHostingView._willUpdateConstraintsForSubtree() + 52
4   SwiftUI                       	    0x7ff915c09264 @objc NSHostingView._willUpdateConstraintsForSubtree() + 21
5   AppKit                        	    0x7ff80c7318ad -[NSView _updateConstraintsForSubtreeIfNeededCollectingViewsWithInvalidBaselines:] + 73
6   AppKit                        	    0x7ff80c731a0e -[NSView _updateConstraintsForSubtreeIfNeededCollectingViewsWithInvalidBaselines:] + 426
7   AppKit                        	    0x7ff80c731771 __45-[NSView updateConstraintsForSubtreeIfNeeded]_block_invoke + 48
8   AppKit                        	    0x7ff80c731542 -[NSView updateConstraintsForSubtreeIfNeeded] + 229
9   AppKit                        	    0x7ff80c75a39e __62-[NSWindow(NSConstraintBasedLayout) updateConstraintsIfNeeded]_block_invoke + 134
10  AppKit                        	    0x7ff80c75a2c2 -[NSWindow(NSConstraintBasedLayout) updateConstraintsIfNeeded] + 253
11  AppKit                        	    0x7ff80c75a105 __NSWindowGetDisplayCycleObserverForUpdateConstraints_block_invoke + 430
12  AppKit                        	    0x7ff80c759f17 NSDisplayCycleObserverInvoke + 155
13  AppKit                        	    0x7ff80c759aa4 NSDisplayCycleFlush + 921
14  QuartzCore                    	    0x7ff810d07fbe CA::Transaction::run_commit_handlers(CATransactionPhase) + 98
15  QuartzCore                    	    0x7ff810d06cdc CA::Transaction::commit() + 376
16  AppKit                        	    0x7ff80c7fb191 __62+[CATransaction(NSCATransaction) NS_setFlushesWithDisplayLink]_block_invoke + 285
17  AppKit                        	    0x7ff80cf42f1c ___NSRunLoopObserverCreateWithHandler_block_invoke + 41
18  CoreFoundation                	    0x7ff809c55500 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
19  CoreFoundation                	    0x7ff809c55392 __CFRunLoopDoObservers + 543
20  CoreFoundation                	    0x7ff809c54824 __CFRunLoopRun + 840
21  CoreFoundation                	    0x7ff809c53e1c CFRunLoopRunSpecific + 562
22  HIToolbox                     	    0x7ff8129045e6 RunCurrentEventLoopInMode + 292
23  HIToolbox                     	    0x7ff812904213 ReceiveNextEventCommon + 283
24  HIToolbox                     	    0x7ff8129040e5 _BlockUntilNextEventMatchingListInModeWithFilter + 70
25  AppKit                        	    0x7ff80c68ef6d _DPSNextEvent + 927
26  AppKit                        	    0x7ff80c68d62a -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1394
27  AppKit                        	    0x7ff80c67fcd9 -[NSApplication run] + 586
28  AppKit                        	    0x7ff80c653c57 NSApplicationMain + 817
29  AlDente                       	       0x1017fc26c 0x1017f1000 + 45676
30  dyld                          	       0x10ada352e start + 462
```

I can provide the full crash report privately if needed.

**Comment by chrisgot** [comment]
I have the same issue on my 15" MBP Late 2016. I already reported it via E-Mail to the support.

**Comment by jeanpascal395** [comment]
I join a similar problem with my 15'' MBP Late 2016. After I enabled the stats in the settings, the app crashes if I click on the icon in die menu bar. Function and display of the charging Icon in the menu bar are not affected. Maybe the crash behavior depends on the stat function?

**Comment by MatthiasKerbl** [comment]
Hi everyone,

Thank you for letting us know about this bug. We are working on a fix for this issue and will release it with one of the next updates.

Best,
Matthias

**Comment by MatthiasKerbl** [comment]
Hi everyone,

We have just released AlDente 1.22.3 which includes a fix for this issue. To get AlDente 1.22.3 you just need to update within the app under Settings/General/Check for Updates.

Please report back if it works for you.

Cheers,
Matthias
